### ISSUE TITLE
Fix dropdown menu behavior and mobile layout

### ISSUE THREAD

**Issue by RobertAnthonyDevelopment** [RobertAnthonyDevelopment/to-do-list-app#1]
### Issue Description
In the To-Do List App, the dropdown menus (like 'Recurring') stay open after being clicked and do not close when tapping elsewhere. Additionally, the layout looks cramped on mobile devices (see the attached screenshot).

### Steps to Reproduce
1. Open the app on a mobile device or emulator.
2. Tap on the 'Recurring' dropdown menu.
3. Notice that the menu stays open.
4. Check the UI elements – they appear misaligned or cramped on small screens.

### Expected Behavior
- Dropdown menus should open and close as expected.
- Tapping outside a menu should close it.
- UI elements should adjust properly for mobile screens.

### Actual Behavior
- Menus stay open after being clicked.
- Layout looks cramped or misaligned on small screens.

### Suggested Fix
- Use `onDismissRequest` for dropdowns to close on outside clicks.
- Add `verticalScroll()` to support mobile scrolling.
- Apply `Modifier.fillMaxWidth()` to align elements.

### Environment
- Device: Your Device
- OS: Your OS
- Branch: bugfix/fix-recurring-dropdown-menu

**Screenshot:**  
(Attach the screenshot here)

**Comment by RobertAnthonyDevelopment** [comment]
# Update @13:47 PM PST
The dropdown menu has been successfully reformatted, and recurring tasks can now be added dynamically as expected. However, there's still an outstanding issue where the dropdown does not close upon selection of a weekly task. I'll be addressing this by implementing a solution to ensure the menu properly dismisses when an external tap occurs. I'll push the fix shortly via a pull request for review.
![Screenshot 2024-10-21 at 1 48 33 PM](https://github.com/user-attachments/assets/6411cb97-51fd-4bc2-89f8-5bb95af0fea5)

**Comment by RobertAnthonyDevelopment** [comment]
### Update:  
The latest changes have been pushed to the `bugfix/fix-recurring-dropdown-menu` branch. These changes improve the stability of the dropdown menu and allow recurring tasks to be added dynamically.

### Remaining Issue:  
- The dropdown still doesn't close upon selecting a weekly task.

### Next Steps:  
I will continue working on the branch and push further updates to fully resolve the issue. Once the fix is complete and tested, it will be merged into `main`.
